### DISCUSSION TITLE
Installation threw exception then failed

### DISCUSSION THREAD

**Discussion by DcPublished** [highfestiva/finplot#299] [Q&A] [answered]
Hi.   During finplot installation, it threw the error below and then aborted: 

`error: metadata-generation-failed`

From the log, it appears that it failed on this line: 
> File "/private/var/folders/7h/q3q6x4vs7n9gn0mdsm3d51xw0000gn/T/pip-build-env-j9i4jpsu/overlay/lib/python3.10/site-packages/pyqtbuild/builder.py", line 67, in apply_user_defaults
>           raise PyProjectOptionException('qmake',
>       sipbuild.pyproject.PyProjectOptionException
>       [end of output]

Note that there is no ".../site-packages/**pyqtbuild**/..." folder.

I tried several times but could not resolve this.   Has anyone else seen this?   Any suggestions on fixing this?  

Thanks,

Dan.

p.p.s. I'm using a new Mac Pro Max.   I have 30 years of experience as a software engineer on Windows, but I just started using a Mac about six weeks ago, so this might be a user error.   Any pointers are gratefully accepted.

**Comment by DcPublished** [comment]
FinPlot is a nice library, but it's not natively compatible with a Mac M1. I got through the installation issues, but so far, the example plot code throws an error.  

At this point, I have neither the time nor energy to try to these issues.  So I'm moving to another plotting library like mplfinance or Seaborn.   

Regards,

Dan. 

p.s. Some details: 

After reviewing the Finplot installation instructions, I noticed that it required PyQt5.   When I tried to installed that library, the install failed.   After research, I found a solution: 

```
brew install pyqt5
echo 'export PATH="/opt/homebrew/opt/qt@5/bin:$PATH"' >> ~/.zshrc
echo 'export PATH="/opt/homebrew/opt/pyqt@5/5.15.4_1/bin:$PATH"' >> ~/.zshrc
source ~/.zshrc
```

This installed PyQt5, but under Rosetta installation was very slow.   What's worse is that the example code throws an error: 
```
fplt.candlestick_ochl(df[['Open', 'Close', 'High', 'Low']])
fplt.show()
```
I suspect that this is related to the difference between Intel Macs and M1 Macs, or between Macs and Intel machines.

**Comment by DcPublished** [comment, marked as answer]
Before heading to bed, I tried on last time and found a solution.   If the file is executed in Pycharm's Debug mode, it fails as described above.  When executed in "Run" mode, it works.   When executed from Terminal, it works.   

I have to dig into this further.

Dan.

**Reply by DcPublished** [reply]
After working with the example code further, I'm pleased with FinPlot.   It exceeds my expectations and requirements.

Regards,

Dan.

**Comment by highfestiva** [comment]
Hi Dan,

Unfortunately I don't know Mac myself, and there is no other regular maintainer than me. If you'd like to become one, you're more than welcome!

KR,
Jonas
